core/server/api/v0.1/decorators/urls.js

```diff
@@ -0,0 +1,86 @@
+const urlService = require('../../../services/url');
+const {urlFor, makeAbsoluteUrls} = require('../../../services/url/utils');
+
+const urlsForPost = (post, options) => {
+    post.url = urlService.getUrlByResourceId(post.id);
+
+    if (options.columns && !options.columns.includes('url')) {
+        delete post.url;
+    }
+
+    if (options && options.context && options.context.public && options.absolute_urls) {
+        if (post.feature_image) {
+            post.feature_image = urlFor('image', {image: post.feature_image}, true);
+        }
+
+        if (post.og_image) {
+            post.og_image = urlFor('image', {image: post.og_image}, true);
+        }
+
+        if (post.twitter_image) {
+            post.twitter_image = urlFor('image', {image: post.twitter_image}, true);
+        }
+
+        if (post.html) {
+            post.html = makeAbsoluteUrls(post.html, urlFor('home', true), post.url).html();
+        }
+
+        if (post.url) {
+            post.url = urlFor({relativeUrl: post.url}, true);
+        }
+    }
+
+    if (options && options.withRelated) {
+        options.withRelated.forEach((relation) => {
+            if (relation === 'tags' && post.tags) {
+                post.tags = post.tags.map(tag => urlsForTag(tag, options));
+            }
+
+            if (relation === 'author' && post.author) {
```

Comment:
The `author` use is deprecated, but still added it for compatibility and probably will have to be removed along with the rest of places where it's being used

Reply:
Only copy paste yes.

Reply:
As discussed. We agree having hardcoded relations in here is not the nicest way to solve this problem, but in a given situation, it's the only option to make it work. Will be revisited once v2 is introduced.